.github/workflows/main.yml

```diff
@@ -146,7 +146,7 @@ jobs:
       - run: make lint
         shell: bash
       - run: make mypy
-        if: ${{ !cancelled() }}
+        if: ${{ !cancelled() && matrix.os != 'windows-latest' }}
```

Comment:
Dont we want to do this either if the run is not cancelled `or` if the os is not windows (in general the workflow will not be cancelled and even if it is we always want to skip i.e run if not cancelled so it needs to be an `or` not `and` I think)